src/motor_node.cc

```diff
@@ -249,25 +249,42 @@ int main(int argc, char* argv[]) {
     float minCycleTime = 0.75 * expectedCycleTime;
     float maxCycleTime = 1.25 * expectedCycleTime;
 
-
     // Clear any commands the robot has at this time
     robot->clearCommands();
 
-    last_time = ros::Time::now();
-    ros::Time last_sys_maint_time = last_time;
+    last_loop_time = ros::Time::now();
+    ros::Time last_sys_maint_time = last_loop_time;
     ros::Duration elapsed_since_last_action;
 
+    double leftWheelVelocity  = 0.0;
+    double leftWheelPosition  = 0.0;
+    double leftLastWheelPos   = 0.0;
+    double rightWheelVelocity = 0.0;
+    double rightWheelPosition = 0.0;
+    double rightLastWheelPos  = 0.0;
+    robot-> getWheelJointPositions(leftLastWheelPos, rightWheelPosition);
+    ctrlLoopDelay.sleep(); // We do this delay so velocity calc uses non-zero time
+
     ROS_WARN("Starting motor control node now");
 
     // Implement a speed reset while ESTOP is active and a delay after release
     double estopReleaseDeadtime = 0.8;
     double estopReleaseDelay    = 0.0;
     while (ros::ok()) {
         current_time = ros::Time::now();
-        elapsed = current_time - last_time;
-        last_time = current_time;
-        robot->readInputs();
+        elapsed = current_time - last_loop_time;
+        last_loop_time = current_time;
         float elapsedSecs = elapsed.toSec();
+
+        // Determine and set measured wheel velocities in rad/sec from hardware positions in radians
```

Comment:
I did not put it in readInputs because it requires history of elapsedSecs.

After seeing this value jump I am now thinking I want even greater time spacing due to our crappy encoder counts.
By the way encoder ticks are getting even worse with non-geared avalon we go to a whopping 86 ticks/rev. Good lord.

So with your comment and seeing the velocity jump a bit even now I am thinking of spacing out even more the measurement to 250ms or even 500ms.  It will be more accurate at low speeds and we have zero hope in hell of having it be of much value at all if it is calculated very ODOM report from the MCB.

So it could be in ODOM message in readInputs() or in every readInputs() but I do insist it requires a timer and I am not real happy about that timer being in MotorHardware as it's hard to support being dependent on ODOM report rate so if MCB fails velocity could get stuck at some high value.

Reply:
Have pushed changes so that updates to 250msec and it is much more stable so that fixes earlier where it was updated too recently and encoder resolution was making the joint velocity have significant jitter.